vuepress/docs/next/tutorials/ecr/github-actions-workflow-from-ecg.md

```diff
@@ -0,0 +1,219 @@
+       run: sudo /etc/init.d/mysql stop
+     - name: Run backend test
+       run: |
+         chmod +x mvnw
+         ./mvnw -ntp clean verify
+     - name: Package application
+       run: ./mvnw -ntp package -Pprod -DskipTests
+```
+
+## Refine the Workflow definition
+
+- Below are the key and the name of the Job, respectively:
+``` yaml
+ pipeline
+   name: ci pipeline
+```
+The key can be reused in the file as a reference, while the UI displays the Job using the name. By default, the name is defined as the key. For convenience, rename it to `backend`.
+
+- The Workflow is triggered each time a new commit is added or when a pull request is open:
+``` yaml
+on: [push, pull_request]
+```
+However, you can [customize the triggering events](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows). 
+
+- Remove the statement below to ensure the Workflow is not skipped:
+``` yaml
+ if: "!contains(github.event.head_commit.message, '[ci skip]') &&  
+ !contains(github.event.head_commit.message, '[skip ci]') && 
+ !contains(github.event.pull_request.title, '[skip ci]') && 
+ !contains(github.event.pull_request.title, '[ci skip]')"
+```
+GitHub [recently announced](https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/) a baseline feature where the Workflow is skipped if the pull request title or latest commit message contains either `skip ci` or `ci skip`.
+
+- To avoid conflicts when starting any other MySQL database (e.g. through containers), use the below action to stop the MySQL server pre-installed on the Ubuntu runners:
+``` yaml
+name: Stop MySQL server 
+```
+
+
+## Add configuration for micro frontends
+
+### Prerequisites
+For this part of the tutorial, you will need the code for both the backend and the micro frontends. The micro frontends can be generated with the ECG or created manually.
+
+### Start defining a new Job
+
+To create a Job definition specific to a micro frontend:
+1. Add a new entry at the same indentation level as the previous Job (e.g `pipeline` or `backend` from the last section)
+2. Define the name (e.g. `microfrontends`)
+3. Define the runner using the `runs-on` property (e.g. `ubuntu-latest`)
+4. Checkout the project and install npm 
+5. From your micro frontend root folder install dependencies and run the tests:
+``` yaml
+steps:
+ - uses: actions/checkout@v2
+ - uses: actions/setup-node@v2.1.4
+   with:
+     node-version: '14.15.0'
+ - name: Run tests
+   run: |
+     cd <mfe_path>
+     npm install
+     npm test
+```
+
+The final Job definition looks like this:
+``` yaml
+micro-frontends:
+ name: micro frontend job
+ runs-on: ubuntu-latest
+ steps:
+   - uses: actions/checkout@v2
+   - uses: actions/setup-node@v2.1.4
+     with:
+       node-version: '14.15.0'
+   - name: Run tests
+     run: |
+       cd <mfe_path>
+       npm install
+       npm test
+```
+
+### Use a matrix strategy for a Job with multiple micro frontends
```

Comment:
To shorten for the left nav to look neat?

Matrix strategy for Job with many MFEs

Reply:
aha! i was wondering if the left nav is a direct representation of the header. yes for sure